docs/FIFOMarket.md

```diff
@@ -1,178 +1,111 @@
 ## FIFOMarket
 
-
-
-
-
-
-
 ---
 
-### _erc1820
+### \_erc1820
 
 ```solidity
 contract IERC1820RegistryUpgradeable _erc1820
 ```
 
-
-
-
-
-
-### _removal
+### \_removal
 
 ```solidity
 contract Removal _removal
 ```
 
-
-
-
-
-
-### _certificate
+### \_certificate
 
 ```solidity
 contract Certificate _certificate
 ```
 
-
-
-
-
-
-### _bridgedPolygonNori
+### \_bridgedPolygonNori
 
 ```solidity
 contract BridgedPolygonNORI _bridgedPolygonNori
 ```
 
-
-
-
-
-
-### _queue
+### \_queue
 
 ```solidity
 mapping(uint256 &#x3D;&gt; uint256) _queue
 ```
 
-
-
-
-
-
-### _queueHeadIndex
+### \_queueHeadIndex
 
 ```solidity
 uint256 _queueHeadIndex
 ```
 
-
-
-
-
-
-### _queueNextInsertIndex
+### \_queueNextInsertIndex
 
 ```solidity
 uint256 _queueNextInsertIndex
 ```
 
-
-
-
-
-
-### _noriFeeWallet
+### \_noriFeeWallet
 
 ```solidity
 address _noriFeeWallet
 ```
 
-
-
-
-
-
-### _noriFee
+### \_noriFee
 
 ```solidity
 uint256 _noriFee
 ```
 
-
-
-
-
-
 ### initialize
 
 ```solidity
 function initialize(address removalAddress, address bridgedPolygonNoriAddress, address certificateAddress, address noriFeeWalletAddress, uint256 noriFee) public
 ```
 
-
-
-
-
-
-### _queueLength
+### \_queueLength
 
 ```solidity
 function _queueLength() private view returns (uint256)
 ```
 
-
-
-
-
-
 ### numberOfNrtsInQueue
 
 ```solidity
 function numberOfNrtsInQueue() public view returns (uint256)
 ```
 
-
-
-
-
-
 ### onERC1155BatchReceived
 
 ```solidity
 function onERC1155BatchReceived(address, address, uint256[] ids, uint256[], bytes) public returns (bytes4)
 ```
 
-
-
-
-
-
 ### tokensReceived
 
 ```solidity
 function tokensReceived(address, address, address, uint256 amount, bytes userData, bytes) external
 ```
 
-
-
 _Called automatically by the ERC777 (nori) contract when a batch of tokens are transferred to the contract._
 
-
-
-### supportsInterface
+### \_addActiveSupplier
 
 ```solidity
-function supportsInterface(bytes4 interfaceId) public view virtual returns (bool)
+function addActiveSupplier(address, supplierAddress) external
 ```
 
+_Called when a new supplier is added to the marketplace. If the first supplier, initializes a cicularly doubly-linked list, where initially the first supplier points to itself as next and previous. When a new supplier is added, at the position of the current supplier, update the previous pointer of the current supplier to point to the new supplier, and update the next pointer of the previous supplier to the new supplier._
```

Comment:
These are autogenerated by `hardhat docgen`, we shouldn't edit these files manually as they'll be overwritten